adam/experiment/__init__.py

```diff
@@ -197,7 +223,20 @@ def execute_experiment(
             except OSError:
                 logging.warning("Cannot log learner state to %s", str(learner_path))
                 log_learner_state = False
-                logging.warning("Proceeding without logging learner state")
+                logging.warning(
+                    "Proceeding without logging learner state. The experiment results from the observers won't be valid if this is not logged."
+                )
+        observer_path = learner_logging_path / "observer_state"
+        if not observer_path.exists():
+            try:
+                observer_path.mkdir()
+            # if we don't have a directory where we can log our observer state, we simply don't log it
+            except OSError:
+                logging.warning("Cannot log observer state to %s", str(observer_path))
+                log_learner_state = False
+                logging.warning(
+                    "Proceeding without logging learner state. The experiment results from the observers won't be valid if this is not logged."
```

Comment:
Slightly confusing for similar reasons; here I think the "this" is supposed to refer to the observer state? So maybe something like this:
```suggestion
                    "Proceeding without logging learner. The experiment results from the observers won't be valid if the experiment is resumed without a log of the observer state."
```